libraries/redcore/api/soap/operation/operation.php

```diff
@@ -61,63 +55,65 @@ public function readList(
 			$dataGet = JArrayHelper::fromObject($dataGet);
 		}
 
-		if ($limitStart != 0)
-		{
-			$dataGet['list']['limitstart'] = (int) $limitStart;
-		}
-
-		if ($limit != 20)
-		{
-			$dataGet['list']['limit'] = (int) $limit;
-		}
-
-		if (!is_null($filterSearch))
-		{
-			$dataGet['filter']['search'] = $filterSearch;
-		}
+		$dataGet['list']['limitstart'] = (int) (isset($data->limitStart) ? $data->limitStart : 0);
+		$dataGet['list']['limit'] = (int) (isset($data->limit) ? $data->limit : 20);
+		$dataGet['filter']['search'] = (string) (isset($data->filterSearch) ? $data->filterSearch : '');
 
-		if (!empty($filters) && is_array($filters))
+		if (isset($this->webservice->configuration->operations->read->list->fields))
```

Comment:
this is unnecessary, you already have this function in the hal. Just use that function to get filter fields